src/canvas.py

```diff
@@ -87,16 +96,17 @@ def set_axis_properties(self):
         self.axis.set_xscale(plot_settings.xscale)
 
     def set_ticks(self):
-        bottom = pyplot.rcParams["xtick.bottom"] == "True"
-        left = pyplot.rcParams["ytick.left"] == "True"
-        top = pyplot.rcParams["xtick.top"] == "True"
-        right = pyplot.rcParams["ytick.right"] == "True"
-        minor = pyplot.rcParams["xtick.minor.visible"] == "True"
-        for axis in [self.top_right_axis,
-                     self.top_left_axis, self.axis, self.right_axis]:
-            axis.tick_params(bottom=bottom, left=left, top=top, right=right)
-            if minor:
-                axis.minorticks_on()
+        bottom = pyplot.rcParams["xtick.bottom"]
+        left = pyplot.rcParams["ytick.left"]
+        top = pyplot.rcParams["xtick.top"]
+        right = pyplot.rcParams["ytick.right"]
+        for axis in [self.top_right_axis, self.axis]:
+            if pyplot.rcParams["xtick.minor.visible"]:
+                ticks = "both"
+            else:
+                ticks = "major"
```

Comment:
Might want to move ticks out of the loop

Reply:
Yes, that's a miss from my part. In my had it made sense to check the ticks on each axis individually, but that's not needed of course since this parameter is a central one that's not axis dependent.